src/Vocup/IO/Vhf2/Vhf2Format.cs

```diff
@@ -0,0 +1,153 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.IO.Compression;
+using System.Text;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Vocup.Models;
+
+namespace Vocup.IO.Vhf2;
+
+public class Vhf2Format : BookFileFormat
+{
+    private const string headerFileName = "VOCUP VOCABULARY BOOK";
+    private const string bookFileName = "book.2.json";
+    private readonly JsonSerializerOptions options;
+    private readonly Version fileVersion;
+
+    public static Vhf2Format Instance { get; } = new();
+
+    private Vhf2Format()
+    {
+        options = new JsonSerializerOptions
+        {
+            PropertyNamingPolicy = JsonNamingPolicy.CamelCase,
+            WriteIndented = true
+        };
+        options.Converters.Add(new PracticeModeConverter());
+        options.Converters.Add(new PracticeResultConverter());
+
+        fileVersion = new Version(2, 0);
+    }
+
+    public void Read(FileStream stream, VocabularyBook book)
+    {
+        try
+        {
+            using ZipArchive archive = new(stream, ZipArchiveMode.Read, leaveOpen: true);
+
+            ZipArchiveEntry versionFile = archive.GetEntry(headerFileName)
+                ?? throw new VhfFormatException(VhfError.InvalidVersion);
+
+            Version? version;
+
+            using (StreamReader reader = new(versionFile.Open(), Encoding.UTF8))
+            {
+                string? line = reader.ReadLine();
+                if (line == null || !Version.TryParse(line, out version))
+                    throw new VhfFormatException(VhfError.InvalidVersion);
+            }
+
+            if (version.Major > fileVersion.Major) throw new VhfFormatException(VhfError.UpdateRequired);
```

Comment:
I compared the max supported file version. However, that was not a very clever idea either. Now I show _update required_ when the JSON book does not exist and attempt to open it otherwise showing a warning if its version is newer than the max supported version.